mongorestore/filepath_test.go

```diff
@@ -112,6 +112,47 @@ func TestCreateAllIntents(t *testing.T) {
 	})
 }
 
+func TestCreateAllIntentsLongCollectionName(t *testing.T) {
+	// This tests creates intents based on the test file tree:
+	//   testdata/longcollectionname
+	//   testdata/longcollectionname/db1
```

Comment:
I think we have to keep all the long collection name tests separate. Both the unit tests and integration tests use the same `testdata/testdirs/` directory, so if we're going to put the truncated bson/metadata files in another spot, that's going to affect how we do both types of tests.

It's kind of annoying with respect to code duplication, but there's no easy way to otherwise test functions like `CreateAllIntents()` that take a single directory.